### PR TITLE
feat(buttons): integration with NgModelController.$render()

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
@IgorMinar I'm sending first PR for a directive to be used during the workshop. Starting with simpler directives so we've got an occasion to discuss files layout, format, level of complexity etc. A PR will allow us to have discussion based on the source code.

Also, I'm adding you as a collaborator to this repo so you can push stuff if needed.

### CODE REVIEW

File: src/ngmodelctrl_buttons/demo/buttons-checkbox.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      //model -> UI
+      ngModelCtrl.$render = function () {
+        element.toggleClass('active', angular.equals(ngModelCtrl.$modelValue, getTrueValue()));
+      };
+
+      //ui->model
+      element.on('click', function () {
+        scope.$apply(function () {
+          ngModelCtrl.$setViewValue(element.hasClass('active') ? getFalseValue() : getTrueValue());
+          ngModelCtrl.$render();
```
Comment (IgorMinar) [inline src/ngmodelctrl_buttons/demo/buttons-checkbox.js:28]: do we also need to render here? that sounds weird

**Comment by IgorMinar** [comment]
This looks good. The two examples are a bit too similar, but with simple stuff like this it's kind of hard to make them much more different.

**Comment by mgcrea** [comment]
edit: @pkozlowski-opensource explained the workshop purpose so I'll let you guys work quietly!

File: src/ngmodelctrl_buttons/demo/buttons-checkbox.js (hunk 1 above)
Comment (pkozlowski-opensource) [inline src/ngmodelctrl_buttons/demo/buttons-checkbox.js:28]: my impression is that calling $setViewValue won't invoke $render, but going to re-test it.

**Comment by pkozlowski-opensource** [comment]
@IgorMinar yep, those examples are quite simple and similar but I think that this is rather good for a workshop. I'm also preparing exercises to show how to work with parsers and formatters so people are going to have 2 little exercises based on NgModelController.

File: src/ngmodelctrl_buttons/demo/buttons-checkbox.js (hunk 1 above)
Comment (pkozlowski-opensource) [inline src/ngmodelctrl_buttons/demo/buttons-checkbox.js:28]: OK, so what is going on here is that `$render` is only called upon detecting model changes. Calling $setViewValue is going to update the model, but $render won't fire due to this check:
https://github.com/angular/angular.js/blob/master/src/ng/directive/input.js#L1153

In short: as the current AngularJS implementation stands we need to call `$render()` here.

**Comment by pkozlowski-opensource** [comment]
Landed as f3f46b1d58cd7a2b98720f806ed281c1b9d986b4

**Pull request closed without merging**